### ISSUE TITLE
Create new Model transformer classes

### ISSUE THREAD

**Issue by awaterman5** [WaterwayData/FeatureSpace#26]
* This should act as the transformer for all ML model-related functions:
1. train(): Input = FeatureSet (like training data) --> Output = a new model file saved in /data/models/model_[name]_[date].dll (prior to running train or load, this model object is just metadata), and store a list of input FeatureSet + output model file versions used to train
2. load(): Input = filename or label of the model instance to use (and if the label is provided, find the latest version)
3. initialize(): Input = set of hyperparameters to load into memory for this model instance (but be careful about changing hyperparams on a given model file)
4. predict(): Input = FeatureSet (like test data) --> Output = a set of predictions stored as a new FeatureSet (also store a list of all input FeatureSet + model file + output FeatureSet versions)
5. evaluate(): Input = FeatureSet with labeled cols --> Output = a set of metrics detailing the performance of the latest set of predictions vs. the labeled cols in the input feature set + a new FeatureSet with the comparisons for each data point in the input feature set 
6. save(): Save this model object to file as a pkl/dll file containing the hyperparameters, a reference to the FeatureSet used to train it (including epochs, etc.), when it was trained, and the model object needed to create predictions (i.e. tensorflow, scikit-learn, etc.) 

Each function can be overridden by a child class written specifically for different ML libraries (tensorflow, scikit-learn, etc.) and different types of models (classification, timeseries, regression, clustering, etc.).  Also note this can be used for Unsupervised ML by skipping train() and just using initialize() to set up the params specific to that model.
